src/cloudant/replicator.py

```diff
@@ -69,6 +69,10 @@ def create_replication(self, source_db=None, target_db=None,
         :returns: Replication document as a Document instance
         """
 
+        # WIP: Handle 'repl_id' when 'source' and 'target' parameters exist
+        if not repl_id and source_db and isinstance(source_db, str):
```

Comment:
do you mean `source` here (not `source_db`)?  `source_db` shouldn't be a string according to the docs.

> source_db: Database object to replicate from.  Can be either a ``CouchDatabase`` or ``CloudantDatabase`` instance.

> source: Optional ``str`` or ``dict``

Reply:
also, why are we setting `repl_id = source_db`? is it just a new default?

Reply:
@smithsz `source_db` should be a Database object, but because of duck typing it will accept a `str`.
Without this code, if you pass in your own replication id it will set to the `source_db` parameter:
```
self.replicator.create_replication(
            repl_id,
            source=self.db.database_url,
            target=self.target_db.database_url)
```